src/routing/comm.rs

```diff
@@ -179,11 +173,29 @@ impl Comm {
         // Run all the sends concurrently (using `FuturesUnordered`). If any of them fails, pick
         // the next recipient and try to send to them. Proceed until the needed number of sends
         // succeeds or if there are no more recipients to pick.
-        let send = |recipient, msg| async move { (self.send_to(recipient, msg).await, recipient) };
-
+        let send = |recipient: (SocketAddr, XorName), mut msg: MessageType| async move {
+            msg.update_dest_info(None, Some(recipient.1));
```

Comment:
I do think make that `DestInfo` an optional field , and only fill it at the constructing place when necessary will be better than here that have to `update` it separately later on.

Reply:
Agreed 👍 But I would like to do this in a separate PR as the changes would span all across the repos.